src/typed-error.ts

```diff
@@ -0,0 +1,29 @@
+class BaseError {
+    constructor () {
+        Error.apply(this, arguments)
+    }
+}
+BaseError.prototype = Object.create(Error.prototype)
```

Comment:
Because `Error` is special in javascript and you can't extend it in the normal way, basically typescript generates:
```js
        function TypedError(message) {
            var _this = _super.call(this) || this;
            ...
            return _this;
        }
```
which when `_super` is Error means that the returned object is an instanceof Error but not an instanceof TypedError, due to returning the unmodifiable Error instance.  When using the BaseError thing of just `Error.apply(this, arguments)` and the prototype assignment then the created object will count as an instanceof both Error and TypedError correctly.  As for doing the prototype assignment/etc on TypedError direct, typescript complains about it due to certain types being optional on Error but required on TypedError